test/components/personkort/PersonkortHeaderTest.tsx

```diff
@@ -37,46 +37,81 @@ describe("PersonkortHeader", () => {
 
   it("viser 'Egenansatt' når isEgenansatt er true fra API", async () => {
     stubEgenansattApi(apiMockScope, true);
-    stubDiskresjonskodeApi(apiMockScope);
     renderPersonkortHeader();
 
     expect(await screen.findByText("Egenansatt")).to.exist;
   });
 
   it("viser ikke 'Egenansatt' når isEgenansatt er false fra API", async () => {
     stubEgenansattApi(apiMockScope, false);
-    stubDiskresjonskodeApi(apiMockScope, "6");
     renderPersonkortHeader();
-    await screen.findByText("Kode 6");
 
     expect(screen.queryByText("Egenansatt")).not.to.exist;
   });
```

Comment:
Vi fjernet litt ting fra noen av testene: feks ga det ikke mening å kalle stub for diskresjonskode når man tester egenansatt?

Reply:
Nei, de er ikke helt bra disse testene. Grunnen til at man gjør `stubDiskresjonskodeApi` og en `await screen.find()` her er at man ønsker å vente til en annen tag blir rendret før man sjekker at Egenansatt-tagen ikke er der. Samme gjelder linje 71 osv.
Man kan evt la det være og legge inn en kommentar på det, eller så kan vi skrive om testene til å ikke bruk api-stubs, men heller bare gjøre sette data direkte på queryclient som vi gjør i de fleste andre tester så slipper vi deale med async rendring i testen.

Reply:
Ja, vi ville jo egt bruke `queryClient`, men gikk for stub siden det var "normalen" i testen 🤷🏼‍♂️
Jeg kan skrive om jeg, er ikke store jobben 🤔